Document the workflow run-as identity, including run as initiator (#38539)
DataDog/documentation: content/en/actions/workflows/trigger.md

- link: "/getting_started/workflow_automation/"
  tag: "Documentation"
  text: "Getting Started with Workflow Automation"
- link: "/actions/workflows/access_and_auth/#workflow-identity"
  tag: "Documentation"

>   text: "Find out more about the identity a workflow runs as"

Reviewer: ```suggestion
  text: "Find out more about the identity used to run a workflow"
```
